server/models/TransactionsImport.ts

```diff
@@ -14,29 +14,65 @@ import sequelize, { DataTypes, Model } from '../lib/sequelize';
 
 import Activity from './Activity';
 import Collective from './Collective';
+import ConnectedAccount from './ConnectedAccount';
 import TransactionsImportRow from './TransactionsImportRow';
 import UploadedFile from './UploadedFile';
 import User from './User';
 
 // For some reason `CreationOptional` is not enough to make fields optional
 type CreationAttributes = InferCreationAttributes<
   TransactionsImport,
-  { omit: 'id' | 'csvConfig' | 'createdAt' | 'updatedAt' | 'deletedAt' | 'UploadedFileId' }
+  {
+    omit:
+      | 'id'
+      | 'settings'
+      | 'data'
+      | 'lastSyncAt'
+      | 'createdAt'
+      | 'updatedAt'
+      | 'deletedAt'
+      | 'UploadedFileId'
+      | 'ConnectedAccountId';
+  }
 >;
 
 export const TransactionsImportTypes = ['CSV', 'MANUAL', 'PLAID'] as const;
 
+type TransactionsImportSettings = {
+  csvConfig?: Record<string, unknown>;
+};
+
+type TransactionsImportData = {
+  lockedAt?: string;
+  plaid?: {
+    lastSyncCursor?: string;
+    syncAttempt?: number;
+    lastSyncErrorMessage?: string;
+  };
+};
+
+// A custom error type for when a transaction import is locked
+export class TransactionsImportLockedError extends Error {
+  constructor() {
+    super('This import is already locked');
+    this.name = 'TransactionsImportLockedError';
+  }
+}
+
 class TransactionsImport extends Model<InferAttributes<TransactionsImport>, CreationAttributes> {
-  public declare id: CreationOptional<number>;
+  public declare id: number;
   public declare CollectiveId: ForeignKey<Collective['id']>;
-  public declare UploadedFileId: CreationOptional<ForeignKey<UploadedFile['id']>>;
+  public declare UploadedFileId: ForeignKey<UploadedFile['id']>;
+  public declare ConnectedAccountId: ForeignKey<ConnectedAccount['id']>;
   public declare source: string;
   public declare name: string;
   public declare type: (typeof TransactionsImportTypes)[number];
-  public declare csvConfig: CreationOptional<Record<string, unknown>> | null;
-  public declare createdAt: CreationOptional<Date>;
-  public declare updatedAt: CreationOptional<Date>;
-  public declare deletedAt: CreationOptional<Date>;
+  public declare settings: TransactionsImportSettings | null;
+  public declare data: TransactionsImportData | null;
+  public declare createdAt: Date;
+  public declare updatedAt: Date;
+  public declare deletedAt: Date;
+  public declare lastSyncAt: Date;
```

Comment:
Removed the `CreationOptional` since they were not doing their job properly. We're using `omit` in `InferCreationAttributes` instead.